### PR TITLE
API-1783: Bump 1.30.1

### PR DESCRIPTION
**Pull request by soltysh**

### CODE REVIEW

**Comment by soltysh** [comment]
/override ci/prow/verify-client-go 
this is expected failure, since I'll update o/api in a followup

File: go.mod [hunk 1]
```diff
@@ -7,8 +7,8 @@ toolchain go1.22.1
 require (
 	github.com/gogo/protobuf v1.3.2
 	golang.org/x/tools v0.18.0
-	k8s.io/api v0.30.0
-	k8s.io/apimachinery v0.30.0
+	k8s.io/api v0.30.1
+	k8s.io/apimachinery v0.30.1
```
Comment (p0lyn0mial) [inline go.mod:11]: all deps match the ones defined in k/k in `1.30.1` exactly :)

File: config/v1/zz_generated.crd-manifests/0000_10_openshift-controller-manager_01_builds.crd.yaml [hunk 2]
```diff
@@ -146,9 +146,15 @@ spec:
                                   description: The key to select.
                                   type: string
                                 name:
-                                  description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
-                                    TODO: Add other useful fields. apiVersion, kind,
-                                    uid?'
+                                  default: ""
+                                  description: 'Name of the referent. This field is
```
Comment (p0lyn0mial) [inline config/v1/zz_generated.crd-manifests/0000_10_openshift-controller-manager_01_builds.crd.yaml:150]: Where did we get this difference from?

Comment (soltysh) [reply config/v1/zz_generated.crd-manifests/0000_10_openshift-controller-manager_01_builds.crd.yaml:150]: Coming from k8s 1.30.1, see the other commit here https://github.com/openshift/api/pull/1906/commits/f4098b528011ac02d84c344c1b3e20f88aeca09f#diff-42206d7b9766bf3f301053e91bdc4d95dbbd501c48bad91d141689177c716b00

**Pull request merged**